Cython/Utility/ImportExport.c

```diff
@@ -163,16 +171,42 @@ static PyObject *__Pyx_ImportDottedModuleRelFirst(PyObject *name, PyObject *part
 
 /////////////// Import.proto ///////////////
 
-static PyObject *__Pyx_Import(PyObject *name, PyObject *from_list, int level); /*proto*/
+static PyObject *__Pyx_Import(PyObject *name, PyObject *from_list, int level, PyObject *qualname); /*proto*/
 
 /////////////// Import ///////////////
-//@requires: ObjectHandling.c::PyObjectGetAttrStr
-//@requires:StringTools.c::IncludeStringH
+//@requires: StringTools.c::IncludeStringH
+//@requires: Builtins.c::HasAttr
+//@requires: ImportLookup
 
-static PyObject *__Pyx_Import(PyObject *name, PyObject *from_list, int level) {
+static PyObject *__Pyx_Import(PyObject *name, PyObject *from_list, int level, PyObject *qualname) {
     PyObject *module = 0;
     PyObject *empty_dict = 0;
     PyObject *empty_list = 0;
+    module = __Pyx__Import_LookupLock(qualname);
+    if (likely(module)) {
+        if (from_list) {
+            int module_ready = 1;
+            for (Py_ssize_t i = 0; i < __Pyx_PyList_GET_SIZE(from_list); i++) {
+                PyObject *item = __Pyx_PyList_GetItemRef(from_list, i);
+                if (unlikely(!item)) {
+                    Py_DECREF(module);
+                    return NULL;
+                }
+                int hasattr = __Pyx_HasAttr(module, item);
```

Comment:
```suggestion
                int hasattr = PyObject_HasAttr(module, item);
```

This'd probably be better to not worry about error handling